osu.Game.Rulesets.Mania/Mods/ManiaModHidden.cs

```diff
@@ -2,15 +2,129 @@
+            ManiaPlayfield maniaPlayfield = (ManiaPlayfield)drawableRuleset.Playfield;
+
+            foreach (Column column in maniaPlayfield.Stages.SelectMany(stage => stage.Columns))
+            {
+                HitObjectContainer hoc = column.HitObjectArea.HitObjectContainer;
+                Container hocParent = (Container)hoc.Parent;
+
+                hocParent.Remove(hoc);
+                hocParent.Add(new BufferedContainer
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        hoc,
+                        new LaneCover
+                        {
+                            Coverage = 0.5f,
+                            RelativeSizeAxes = Axes.Both
+                        }
+                    }
+                });
+            }
+        }
+
+        private class LaneCover : CompositeDrawable
+        {
+            private readonly Box gradient;
+            private readonly Box filled;
+            private bool reversed;
+            private readonly IBindable<ScrollingDirection> scrollDirection = new Bindable<ScrollingDirection>();
+
+            public LaneCover()
+            {
+                Blending = new BlendingParameters
+                {
+                    RGBEquation = BlendingEquation.Add,
+                    Source = BlendingType.Zero,
+                    Destination = BlendingType.One,
+                    AlphaEquation = BlendingEquation.Add,
+                    SourceAlpha = BlendingType.Zero,
+                    DestinationAlpha = BlendingType.OneMinusSrcAlpha
+                };
+
+                InternalChildren = new Drawable[]
+                {
+                    gradient = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        RelativePositionAxes = Axes.Both,
+                        Height = 0.25f
+                    },
+                    filled = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both
+                    }
+                };
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(IScrollingInfo configManager)
+            {
+                scrollDirection.BindTo(configManager.Direction);
+                scrollDirection.BindValueChanged(onScrollDirectionChanged, true);
+            }
+
+            private void updateCoverage()
+            {
+                filled.Anchor = reversed ? Anchor.BottomLeft : Anchor.TopLeft;
+                filled.Origin = reversed ? Anchor.BottomLeft : Anchor.TopLeft;
+                filled.Height = coverage;
+
+                gradient.Y = reversed ? 1 - filled.Height - gradient.Height : coverage;
+                gradient.Colour = ColourInfo.GradientVertical(
+                    Color4.White.Opacity(reversed ? 0f : 1f),
+                    Color4.White.Opacity(reversed ? 1f : 0f)
+                );
```

Comment:
thinking out loud here but maybe this could be simplified if instead of doing all this you just rotated the `LaneCover` 180 degrees whenever necessary.

Reply:
...i feel stupid now